<!DOCTYPE html>
<html lang="en">

<head>

    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, shrink-to-fit=no, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">

    <title>Gene Regulatory Network</title>
    <!-- Bootstrap Core CSS -->
    <link href="css/bootstrap.min.css" rel="stylesheet">

    <!-- Custom CSS -->
    <link href="css/simple-sidebar.css" rel="stylesheet">

</head>
<body>
        <!-- Page Content -->
        <div id="page-content-wrapper">
            <div class="container-fluid" style="">
              <header style="">
              <h3> SC462 Elements of Synthetic Biology Spring 2017 </h3>
              <h1 > Gene Regulatory Networks </h1>
            </header>
              <nav class="navbar navbar-default navbar-fixed-top">
  <div class="container"><ul class="nav navbar-nav nav-tabs  nav-justified">
     <li class="active"><a href="#ppt">Project Presentation</a></li>
     <li><a href="#report">Project Report</a></li>
     <li><a href="#source">Source Code</a></li>
     <li><a href="#links">Video Links</a></ul>

  </div>
  </nav>
    <div class="container">
                <div class="row" id="ppt">
                    <div class="col-lg-12" >
                        <h1>Project Presentation</h1>
                        <iframe src="https://docs.google.com/presentation/d/1zsWztE2hZ6IrDa_X6Z3UNm1L0VKYgLTa6UgAaRiYYl0/embed?start=false&loop=false&delayms=3000"
                        frameborder="0" width="900" height="500" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true" style="margin:auto"></iframe>
                    </div>
                </div>
                <div class="row" id="report">
                    <div class="col-lg-12" >
                        <h1>Project Report</h1>
                        <h3>Introduction</h3>
                        <p>Gene Regulation Gene regulation is the mechanism by which the production
of specic gene products(protein nor RNA) is increased or decreased in a cell.
Developmental pathways can be triggered by modifying the gene expression pat-
terns of the cells. All points of gene expression can be regulated. Transcription,
RNA processing and transport, translation and post-translational modication
of a protein, and mRNA degradation are the parts of gene regulation. Regula-
tion of Transcription: A cell determines when a gene is transcribed by regulating
transcription. A cell increases or decreases amount of mRNA generated from
each gene. There are ve dierent ways that RNA polymerase (the enzyme
that makes RNA from a DNA template) can be aected: 1. The specicity of
RNA polymerase for promoters can be changed by a cell using specicity fac-
tors. These regulatory proteins will make RNA polymerase more or less likely
to bind to a promoter. 2. Repressor proteins can bind to an upstream coding
sequence, known an operator, to inhibit gene transcribing by RNA polymerase..
This leads to decreased expression of a gene. 3. Activator proteins can bind
to the operator region to increase the anity of RNA polymerase towards the
promoter. 4. Regulatory regions that lie far upstream to their target gene are
called enhancer regions. Activator binding to an enhancer leads the DNA to
form a loop, which puts the enhancer next to its target gene. This creates in-
creased expression of the target gene. 5. To completely turn o expression of
a gene, cell uses silencer regions. The gene is silenced when a silence region is
bound by transcription factors. This implies that there is no expression of the
gene.
</p>
<h3> Gene Regulation in Prokaryotes</h3>
<p>[5]Regulatory proteins carry the signals to the genes. Regulatory proteins are of
two types. They are positive regulators called activators and negative regulators
called repressors. These activators and repressors are DNA binding proteins.
N
egative Regulators or Repressors The repressor or inhibitor protein binds to
the operator on DNA. The RNA polymerase enzyme is blocked by the repressor
from binding to the promoter which prevents transcription. The repressor binds
to the site and overlaps the polymerase enzyme. This turns o the activity of
the genes. This is called negative control mechanism. An anti-inhibitor or anti-
repressor called inducer inactivates the repressor and thus activates the genes.
This is demonstrated by lactose operon
P
ositive Regulators or Activators The activator allows polymerase enzyme to
bind to the promoter thereby causing activation of transcription by the pro
moter.</p>
<img src="" />
<h4>2.1 Operon</h4>
<p>In bacteria cistrons or structural genes produce enzymes of a metabolic path-
way. These are organised in a cluster which have related fucntions. Poly-
cistronic genes of prokaryotes plus their regulatory genes constitute a system
called operon. Operon is a unit of expression and regulation.</p>
<p>
  <h5>1.Lactose Operon or Lac Operon</h5>
This is a negative control mechanism. The synthesis of enzyme (3-galactosidase
has bee5n studied in detail. This enzyme breaks down lactose into glucose and
galactose. -galactosidase is present in minimal amounts in the absence of lactose.
As soon as lactose is added from outside, the production of - galactosidase is in-
creased by thousand times. As soon as the lactose is consumed, the production
of the enzyme drops again. The enzymes whose production can be increased
by the presence of the substrate are called inducible enzymes. When lactose
is added to the culture medium of E. coli, it induces the formation of three
enzymes (5-galactosidase, permease and transacetylase, which breaks down lac-
tose into galactose and glucose. The genes, which code for these enzymes form
a cluster and are called cistrons or structural genes. They are transcribed at the
same time into a single mRNA chain, which has codons corresponding to all the
three enzymes. The mRNA transcribed from many genes is called polycistronic.
Regulatory genes control the functioning of structural genes to produce mRNA.
There are three structural genes Z, Y and A, which code for enzymes p-galactosidase,
lac permease and transacetylase respectively. Regulatory genes has Regulator I,
a control gene called operator gene O and promoter P. I gene produces a protein
called repressor or inhibitor. The repressor being active, binds to the operator
O and switches o the operator. Thus, the transcription is stopped.
When the inducer (lactose) is supplied from outside, the inducers binds to the
repressor. The l vbactose changes into allolactase on entering the bacteria. Al-
lolactose changes the shape of the repressor which makes it inactive. It cannot
bind to the operator then. The operator becomes free and is \turned on" which
starts the transcription. The presence of the inducer allows the transcription
of Lac operon, as it is not blocked by the repressor protein. The synthesis of
enzymes due to the presence of specic substrate (lactose) is called induction.
It is also called de-repression.</p>
<p>
<h5>2.Tryptophan Operon</h5>
It is also a negative control system but it also forms a biosynthetic pathway.
It is known as repressible system. It works on the clause that when the amino
acid tryptophan is present, the tryptophan operon need not be activated.
The co-repressor activates the repressor protein and it binds the operator and
switches it \o'. Tryptophan is synthesized in ve steps, where each step re-
quires a specic enzyme. The genes which encodes these enzymes lie next to one
another, called trp B, trp D, trp E, trp C, trp A. Tryptophan operon codes for
ve enzymes for the synthesis of amino acid tryptophan. In repressible system,
the regulatory gene produces a repressor protein, which is normally inactive
and doesn't bind to operator on DNA. The repressor on joining the co-repressor
undergoes conformational changes that activate it and enable it to bind to the
operator. This doesn't allow RNA polymerase enzyme to bind to the promoter.
This is opposite to the situation of lac operon in which the repressor is active
on its own and when it binds to the inducer, loses the anity for the operator.
Operon is regulated by tryptophan and this represses the synthesis of tryp-
tophan. The end product of the metabolic chain stops the synthesis of en-
zymes.This allows the bacteria to synthesize enzymes only if needed. This is
known as feedback repression. In feedback inhibition the end product of a
metabolic pathway serves as an allosteric inhibitor of the rst enzyme of the
metabolic chain. Induction and repression prevent the synthesis of unnecessary
enzymes and save valuable energy.</p>


                    </div>
                </div>
                <div class="row" id="source">
                    <div class="col-lg-12" >
                        <h1>Source Code</h1>
                      <pre>  <code>
import pandas as p
import numpy as n
from scipy.io import loadmat
import math

from sklearn.ensemble import RandomForestRegressor
from sklearn.ensemble import RandomForestClassifier
from sklearn.svm import SVR
from sklearn.model_selection import cross_val_score
from sklearn import preprocessing
from sklearn import linear_model

# reading values of different genes of the yeast from the mat file
data_yeastnames = loadmat('ystnames.mat')

# reading values of the different features from the mat file
data_yeast_features = loadmat('yeast.mat')


# dataframe allows to create a multi dimensional matrix allowing to take combiations of rows and columns from different matrices

# here we are accessing the value of dictionary key called 'yystr' which is necessarily an one-D array
yeast_names = p.DataFrame(data_yeastnames['yystr'])

# here value of key 'Yeast' is accessed which consist of values of different features
yeast_features = p.DataFrame(data_yeast_features['Yeast'])


#we are creating a matrix with features of genes as columns and their names as rows
data_initial = p.DataFrame(data_yeast_features['Yeast'].T, columns=data_yeastnames['yystr'])


data_temp = data_initial


#number_genes consist of number of different genes of yeast data set which is 6178 in number in our case
number_genes = len(data_temp.keys())

network = p.DataFrame(n.zeros(shape=(len(data_temp.keys()), len(data_temp.keys()))), index=data_temp.keys(), columns=data_temp.keys())  #create 6178*6178 matrix with indexes of columns and rows as number oof yeast genes

#traversing i through values 0-6177
for i in range(number_genes):

    keys = list(data_temp.keys())

    #we are selecting each gene one by one to be targetGene gene
    targetGene = keys[i]
    keys.remove(targetGene)

    #del keys[2:number_genes]
    inputGenes = keys   #inut genes consist of all genes except the targetGene gene

    #print(targetGene)
    #print(inputGenes)

    # converting list to matrix
    X = data_temp[inputGenes]

    #converting list to matrix
    y = data_temp[targetGene]


    print('Training for targetGene gene {}: {}'.format(i, targetGene))

    #initialising random forest regression object with number of trees in the forest to be 200
    rfr = RandomForestRegressor(n_estimators=200, oob_score=True, n_jobs=-1)

    #this function builds the forest of 200 trees using inuts as X and Y
    rfr.fit(X, y)

    # print(clf)

    #feature_importances is 6187*6178 matrix consisiting the different weights with which the yeast genes are related to each other
    for i, x in zip(range(rfr.feature_importances_.size), rfr.feature_importances_):
         #print(i, inputGenes[i], x)

         #putting the values of weights into the grnmatrix
         network[targetGene][inputGenes[i]] = x


network.to_csv('yeast_grn.csv')  # putting the data to a file


                        </code></pre>

                    </div>
                </div>
                <div class="row" id="links">
                    <div class="col-lg-12" >
                        <h1>Video Links</h1>
                      <p>201401079 : Saloni Dalal - <a href="https://youtu.be/d0IbHsIWmzc">https://youtu.be/d0IbHsIWmzc</a></p>
                      <P>201401095 : Atrayee Deb - <a href="https://www.youtube.com/watch?v=vA_q3vEIR0w&feature=youtu.be"> https://www.youtube.com/watch?v=vA_q3vEIR0w&feature=youtu.be</a> </p>
                      <p>201401113 : Birva Patel - <a href="https://youtu.be/BmQ3qwSM7Y0"> https://youtu.be/BmQ3qwSM7Y0 </a></p>
                      <p>201401115 : Devanshi Bansal -<a href="https://youtu.be/LYihd_ysG-s"> https://youtu.be/LYihd_ysG-s</a> </p>
                      <p>201401159 : Ruchi Kachhia - <a href="https://youtu.be/7nX_nRiWJzo">https://youtu.be/7nX_nRiWJzo</a> </p>
                      <p>201401219 : Prachi Agarwal - <a href="https://youtu.be/aM5lQ3uwLsk" >https://youtu.be/aM5lQ3uwLsk</a> </p>

                    </div>
                </div>
            </div>
      </div>
    </div>
        <!-- /#page-content-wrapper -->

    <!-- jQuery -->
    <script src="js/jquery.js"></script>

    <!-- Bootstrap Core JavaScript -->
    <script src="js/bootstrap.min.js"></script>

    <!-- Menu Toggle Script -->
    <script>
    $("#menu-toggle").click(function(e) {
        e.preventDefault();
        $("#wrapper").toggleClass("toggled");
    });
    </script>
<style>
  .container{
    padding-right: 15px;
    padding-left: 0px;
    margin-right: auto;
    margin-left: 5px;
    margin-top: 5px;
  }
  #ppt{
    padding-top: 10px;
  }
</style>
</body>

</html>
